test_collections/sdk_tests/support/python_testing/models/python_test_parser.py

```diff
@@ -28,42 +29,113 @@
 KEYWORD_IS_COMISSIONING_INDEX = 0
 BODY_INDEX = 0
 
+TC_FUNCTION_PATTERN = re.compile(r"[\S]+_TC_[\S]+")
+TC_TEST_FUNCTION_PATTERN = re.compile(r"test_(?P<title>TC_[\S]+)")
 
-class PythonParserException(Exception):
-    """Raised when an error occurs during the parser of python file."""
+FunctionDefType = Union[ast.FunctionDef, ast.AsyncFunctionDef]
 
 
-def parse_python_test(path: Path) -> PythonTest:
-    """Parse a single Python test file into PythonTest model.
+def parse_python_script(path: Path) -> list[PythonTest]:
+    """Parse a python file into a list of PythonTest models.
 
-    This will also annotate parsed python test with it's path and test type.
+    This will also annotate parsed python tests with their file path and test type.
 
-    It's expected that the python script files has a class with the same file name,
-    and also the following methods:
-     * desc_[class_name] - This method should return a str with the test description
-     * pics_[class_name] - This method should return a list with the PICS required
-        for the test case
-     * steps_[class_name] - This method should return a list with the step description
-     * test_[class_name] -  This method contains the test logic
+    This method will search the file for classes that inherit from MatterBaseTest and
+    then look for methods with the following patterns to extract the needed information:
+     * test_[test_name] - (required) This method contains the test logic
+     * desc_[test_name] - (optional) This method should return a string with the test
```

Comment:
```suggestion
     * desc_[test_name] - (required) This method should return a string with the test
```
Are we assuming desc_ is required?

Reply:
I was thinking of using that method to identify whether the test case is following the template or not. If a test case doesn't have this method it should not break the parsing, so I'm not sure if this comment should say it's required, since the parsing will work if the method doesn't exist for a test case.